spec/integration/repository_spec.rb

```diff
@@ -0,0 +1,89 @@
+describe PackageProvider::Repository do
+  PackageProvider::Repository.temp_prefix = "pp_integration_tests_#{rand(1000)}"
+  persist_folders_prefix = 'pp_integration_per'
+
+  let(:fake_remote_repo_dir) do
+    File.join(PackageProvider.root, 'spec', 'factories', 'testing-repo')
+  end
+  let(:repo) do
+    PackageProvider::Repository.new(fake_remote_repo_dir)
+  end
+
+  after(:each) do
+    repo && repo.destroy
+  end
+
+  describe '#initialize' do
+    it 'makes local copy' do
+      expect(
+        PackageProvider::Spec::Helpers.test_git_status(repo.repo_root)
+      ).to be true
+    end
+  end
+
+  describe '#clone' do
+    let(:dest_dir) do
+      PackageProvider::Spec::Helpers.get_temp_dir_name(persist_folders_prefix)
+    end
+    after(:each) do
+      FileUtils.rm_rf dest_dir rescue nil
+    end
+
+    it 'clones docs folder from test repo' do
```

Comment:
Isn't 'from test repo` irrelevant?
It's better to highlight `clone only`.
...and see the result test string, generally it concatenate from all describe and context and this it, e.g.' ...#clone clone docs folder from test repo'